server/src/controllers/validation/validateSignUp.js

```diff
@@ -0,0 +1,31 @@
+const joi = require('@hapi/joi');
+const Boom = require('@hapi/boom');
+
+const schema = joi.object({
+  name: joi.string().required(),
+  email: joi
+    .string()
+    .email()
+    .required()
+    .pattern(/(\w*@gazaskygeeks.\w*)/),
+  password: joi
+    .string()
+    .regex(/[A-z0-9]{6,}/)
+    .required(),
+  confirmpassword: joi.ref('password'),
+});
+module.exports = (req, res, next) => {
+  const userData = {
+    name: req.body.name,
+    email: req.body.email,
+    password: req.body.password,
+    confirmpassword: req.body.confirmpassword,
+  };
+
+  schema
```

Comment:
this code should be done in the signup controller
This file should only hold the signup schema, so do also rename the file to `signupSchema.js`